src/bufwrite.c

```diff
@@ -2158,6 +2154,13 @@ buf_write(
 	    nchars += len;
 	}
 
+	if (!buf->b_p_fixeol && buf->b_p_eof)
+	{
+	    // write trailing CTRL-Z
+	    (void)write_eintr(write_info.bw_fd, "\x1a", 1);
+	    nchars++;
+	}
```

Comment:
Should we restrict this only for `fileformat == EOL_DOS`?

Reply:
We are dealing with corner cases here, not really trying to catch common uses cases.  I think it's best to give the user ways to control what is written as much is possible.  It seems that adding a restriction to writing CTRL-Z depending on 'fileformat' makes it more complicated.
How about writing the help for these options, can be a separate section that each of the option entries link to.  "If you want files in Unix format (every line NL terminated:...", "If you want to preserve the fileformat exactly as-is, including final CTRL-Z: ...", "If you want to use an old MS-DOS file in a modern environment, fixing line endings and dropping CTRL-Z: ...".  In that style it should make clear how the options work together.

Reply:
I've updated the document. Could you review it?